### DISCUSSION TITLE
Reading vertices and edges from PLY file

### DISCUSSION THREAD

**Discussion by lewisfish** [pyvista/pyvista#3366] [Q&A] [answered]
Hi, 
I have a PLY file with the following header:

ply
format ascii 1.0
element vertex \(vertex_count)
property float x
property float y
property float z
element edge \(edge_count)
property int vertex1
property int vertex2
end_header

I'm trying to read this with pyvista but can only recover the vertices with all edge data appearingly not read. Is there any way to read in the edge data with pyvista? Alternatively, is there an alternative PLY format (or other file format) that is more suitable for this application in pyvista, as I have complete control over the output of data.

**Comment by banesullivan** [comment, marked as answer]
Does `pyvista.read()` not work? Would you share the file directly so we can reproduce and develop a solution?

**Reply by lewisfish** [reply]
Hi, I've added a truncated version of the file into a [gist](https://gist.github.com/lewisfish/52c9c05648429e8fa3d8ab5801b5becf).
I've tried the file in Meshlab and Blender and both give the correct result.

I''ve been using the following script to try to read and plot the data:
```
import numpy as np
import pyvista as pv
filename = "test.ply"
reader = pv.get_reader(filename)
mesh = reader.read()
edges = mesh.extract_all_edges()
edges.plot()
```

**Reply by banesullivan** [reply]
Hm. I am able to reproduce the issue. It seems that the `vtkPLYReader` that we are using for this does not support edge geometry and only supports triangles/face geometry; See https://vtk.org/doc/nightly/html/classvtkPLYReader.html#details

Further, `meshio` cannot handle this format of the PLY file either giving the following error:

```py
>>> mesh = pv.read_meshio(filename)
Expected `element vertex` or `element face` or `obj_info`, got `element edge 23`
Error: Couldn't read file [...]/test.ply as ply
```

I'm thinking that edges are indeed a part of the PLY format but not many readers have focused on supporting edges as polygons are far more common for this format.

http://paulbourke.net/dataformats/ply/

I'd recommend trying to use a different format like STL if possible until this can be worked out upstream in either VTK or meshio

**Reply by MatthewFlamm** [reply]
The vtk reader also has very specific limitations on data that may reside on the vertices, i.e. point data, if this is relevant.  PyVista currently does not expose the underlying methods for controlling this behavior, but it would be a welcome addition.
